store/src/java-test/com/zimbra/cs/datasource/imap/RemoteFolderTest.java

```diff
@@ -0,0 +1,136 @@
+package com.zimbra.cs.datasource.imap;
+
+import org.easymock.EasyMock;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.easymock.PowerMock;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import com.zimbra.cs.mailclient.imap.ImapConnection;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ ImapConnection.class })
+public class RemoteFolderTest {
+
+    @Before
+    public void setUp() throws Exception {
+        PowerMock.mockStatic(ImapConnection.class);
+    }
+
+    @Test
+    public void testRenameTo() throws Exception {
+        final String originalPath = "Test";
+        final String newPath = "Test_1";
+
+        ImapConnection mockConnection = EasyMock.createMock(ImapConnection.class);
+        RemoteFolder f = new RemoteFolder(mockConnection, originalPath);
+
+        mockConnection.rename(originalPath, newPath);
+        PowerMock.expectLastCall();
+
+        PowerMock.replay(mockConnection);
+
+        f.renameTo(newPath);
+
+        PowerMock.verify(mockConnection);
+    }
+
+    // Since the folder name's are same hence rename method will not be called
+    @Test
+    public void testRenameToGrandChildSameName() throws Exception {
+        final String originalPath = "Test/zimbra/ab";
+        final String newPath = "Test/Zimbra/ab";
+
+        ImapConnection mockConnection = EasyMock.createMock(ImapConnection.class);
+        RemoteFolder f = new RemoteFolder(mockConnection, originalPath);
+
+        PowerMock.replay(mockConnection);
+
+        f.renameTo(newPath);
+
+        PowerMock.verify(mockConnection);
+    }
+
+    // If we update the folder from Test/zimbra/pq to Test/Zimbra_1/pq then the
+    // rename of middle layer folder happens only when loop iterates to Test/Zimbra
+    // Below scenario run's in second iteration
+    @Test
+    public void testRenameToMiddleLayerFolder() throws Exception {
```

Comment:
Is the function name correct? I don't think there is any middle layer folder" in originalPath and newPath; parent and child only...?

Reply:
I wanted it to be referenced from another test method , but modified the test name now as  testRenameToDifferentChildFolderName()